### ISSUE TITLE
Training on Custom Dataset with just Person class

### ISSUE THREAD

**Issue by InternetMaster1** [HRNet/HRNet-Semantic-Segmentation#144]
I wish to train HRNetV2-W18-Small-v2 on a custom dataset with just person class from scratch. How to achieve this?

My dataset would be similar to [Supervisely Person Dataset](https://supervise.ly/explore/projects/supervisely-person-dataset-23304/datasets) with just a single person class mask.

I do not wish to perform transfer learning on the existing pre-trained model containing Cityscapes training, as I am afraid that will decrease my model accuracy, plus that is based on multiple classes.

How to get a blank model file of HRNetV2-W18-Small-v2 for a single person class?

**Comment by sunke123** [comment]
@InternetMaster1 
You can use our pretrained weights on imagenet to initialize the network.

If you have prepared the data, I think that you can change the class_num from 19 to 2 and output the segmentation with 2 channels.

**Comment by InternetMaster1** [comment]
@sunke123 
You mean to say this one from the [Image Classification](https://github.com/HRNet/HRNet-Image-Classification) library?

model | #Params | GFLOPs | top-1 error | top-5 error | Link
-- | -- | -- | -- | -- | --
HRNet-W18-C-Small-v2 | 15.6M | 2.42 | 24.9% | 7.6% | [OneDrive](https://1drv.ms/u/s!Aus8VCZ_C_33gRmfdPR79WBS61Qn?e=HVZUi8)/[BaiduYun](https://pan.baidu.com/s/1tbL45sOS4mXNGgyS4YCQww)(Access Code:bnc9)

This above can be used without any changes as a blank model to train for semantic segmentation?

Just in the config code, we have to change class_num to 2.

Correct?

**Comment by InternetMaster1** [comment]
@sunke123 

My custom dataset just contains the image and the mask, i.e.  just a single person class (similar to supervisely person dataset)

In the [Data Preparation](https://github.com/HRNet/HRNet-Semantic-Segmentation#data-preparation) section, there seems to be support for Cityscapes, LIP and PASCAL-Context datasets

> Data preparation
You need to download the Cityscapes, LIP and PASCAL-Context datasets.
Your directory tree should be look like this:
$SEG_ROOT/data
├── cityscapes
│   ├── gtFine
│   │   ├── test
│   │   ├── train
│   │   └── val
│   └── leftImg8bit
│       ├── test
│       ├── train
│       └── val
├── lip
│   ├── TrainVal_images
│   │   ├── train_images
│   │   └── val_images
│   └── TrainVal_parsing_annotations
│       ├── train_segmentations
│       ├── train_segmentations_reversed
│       └── val_segmentations
├── pascal_ctx
│   ├── common
│   ├── PythonAPI
│   ├── res
│   └── VOCdevkit
│       └── VOC2010
├── list
│   ├── cityscapes
│   │   ├── test.lst
│   │   ├── trainval.lst
│   │   └── val.lst
│   ├── lip
│   │   ├── testvalList.txt
│   │   ├── trainList.txt
│   │   └── valList.txt

Even the train & test codes/experiments/configs are catering to the above three datasets.

Could you please provide some guidance as to the directory structure and experiments yaml config file which can be used for a Dataset similar to supervisely person dataset?

**Comment by umairanis03** [comment]
Hi @InternetMaster1 
Were you able to train HRNet with just person class and background ?

**Comment by GutlapalliNikhil** [comment]
Hi @umairanis03 and @InternetMaster1 ,

Can I know that I need to do for training just one class and background?

Hopping for fast reply.

Thanks

**Comment by zhhezhhe** [comment]
Can I set class_num=1 for training just one class and background?

**Comment by GutlapalliNikhil** [comment]
@zhhezhhe 
No, you need to set the class_num = 2.

**Comment by NikhilChowdary-MCW** [comment]
Hi @InternetMaster1 and Everyone, 

Can I know how to do the transfer learning? Please let me know about it.

**Comment by Arshadoid** [comment]
> @zhhezhhe No, you need to set the class_num = 2.

@GutlapalliNikhil were you able to train on 1 class? I'm trying to understand, so if there are 19 classes + background, and if we want to train on only one class, then the unnecessary labels can be changed to ignore_label. In that case it will be 1 class and background. But in class_num shouldn't we have 1? If previously we had 19 and switching the 18 class to ignore wouldn't make it 1 class only?  I re-trained the model on 1 class and result was only 1 color in test for all images, so I am assuming we should somehow pass background as well, but I can not figure out how.
